site/app/libraries/DiffViewer.php

```diff
@@ -638,6 +643,15 @@ private function getDisplay($lines, $type = self::EXPECTED, $option = self::SPEC
                 $html .= "\t</div>\n";
             }
         }
+        if (count($lines) + $num_blanks > 1000) {
```

Comment:
This doesn't work for carriage returns (carriage returns don't add a new line number but still render as a new line).

Reply:
Carriage returns don't seem to work properly with the DiffViewer at all. If you render the DiffViewer with a bunch of them you get 1 line on the DiffViewer but a huge result box. I think supporting this would require a pretty large refactor of the DiffViewer. I have not seen an autograding config use carriage returns so it's not even expected to have them.